src/data/set/lattice.lean

```diff
@@ -1377,6 +1377,29 @@ begin
   { intros x hz x' hw, exact ⟨x ⊔ x', hs le_sup_left hz, ht le_sup_right hw⟩ }
 end
 
+lemma sInter_prod_sInter_subseteq (f₁ : set (set α)) (f₂ : set (set β)) :
+  ⋂₀ f₁ ×ˢ ⋂₀ f₂ ⊆ ⋂₀ ((λ C : set α × set β, C.1 ×ˢ C.2) '' f₁ ×ˢ f₂) :=
+λ x hx _ ⟨s, hs, h⟩, h ▸ ⟨hx.1 s.1 hs.1, hx.2 s.2 hs.2⟩
+
+lemma sInter_prod_sInter (f₁ : set (set α)) (f₂ : set (set β)) (h₁ : f₁.nonempty)
+  (h₂ : f₂.nonempty) : ⋂₀ f₁ ×ˢ ⋂₀ f₂ = ⋂₀ ((λ C : set α × set β, C.1 ×ˢ C.2) '' f₁ ×ˢ f₂) :=
+begin
+  obtain ⟨s₁, h₁⟩ := h₁,
+  obtain ⟨s₂, h₂⟩ := h₂,
+  exact set.subset.antisymm (sInter_prod_sInter_subseteq f₁ f₂) (λ x h, ⟨λ s hs,
+    (h (s ×ˢ s₂) ⟨(s, s₂), ⟨hs, h₂⟩, rfl⟩).1, λ s hs, (h (s₁ ×ˢ s) ⟨(s₁, s), ⟨h₁, hs⟩, rfl⟩).2⟩),
+end
+
+lemma sInter_prod_set (f₁ : set (set α)) (h₁ : f₁.nonempty) (T : set β) :
+  ⋂₀ f₁ ×ˢ T  = ⋂₀ ((×ˢ T) '' f₁) :=
```

Comment:
`⋂ s ∈ S, f s` is preferred over `⋂₀ (f '' S)`.
```suggestion
  ⋂₀ S ×ˢ t = ⋂ s ∈ S, s ×ˢ t :=
```
and everywhere else.

Reply:
I've made the change - however, I don't know how to do the "clever" proofs.

Reply:
I'll take another look later today unless someone beats me to it :)